File: treatment/src/main/java/com/hartwig/actin/treatment/ctc/InterpretedCohortStatusComparator.java
package com.hartwig.actin.treatment.ctc;

import java.util.Comparator;

import org.jetbrains.annotations.NotNull;

class InterpretedCohortStatusComparator implements Comparator<InterpretedCohortStatus> {

    @Override
    public int compare(@NotNull InterpretedCohortStatus status1, @NotNull InterpretedCohortStatus status2) {
        if (status1.open() == status2.open()) {
            return Boolean.compare(status1.slotsAvailable(), status2.slotsAvailable());
        }

        return Boolean.compare(status1.open(), status2.open());
    }
}

File: trial/src/main/kotlin/com/hartwig/actin/trial/status/InterpretedCohortStatusComparator.kt
package com.hartwig.actin.trial.status

internal class InterpretedCohortStatusComparator : Comparator<InterpretedCohortStatus> {

    override fun compare(status1: InterpretedCohortStatus, status2: InterpretedCohortStatus): Int {
        return if (status1.open == status2.open) {
            compareValues(status1.slotsAvailable, status2.slotsAvailable)
        } else {
            compareValues(status1.open, status2.open)
        }
    }
}